src/attribution/onINP.ts

```diff
@@ -260,6 +262,99 @@ export const onINP = (
     return intersectingLoAFs;
   };
 
+  const getLoAFSummary = (attribution: INPAttribution) => {
+    // Stats across all LoAF entries and scripts.
+    const interactionTime = attribution.interactionTime;
+    const inputDelay = attribution.inputDelay;
+    const processingDuration = attribution.processingDuration;
+    let totalNonForcedStyleAndLayoutDuration = 0;
+    let totalForcedStyleAndLayout = 0;
+    let totalIntersectingScriptsDuration = 0;
+    let numScripts = 0;
+    let slowestScriptDuration = 0;
+    let slowestScriptEntry!: PerformanceScriptTiming;
+    let slowestScriptSubpart!: INPSubpart;
+    const subparts: Partial<
+      Record<INPSubpart, Partial<Record<ScriptInvokerType, number>>>
+    > = {};
+
+    for (const loafEntry of attribution.longAnimationFrameEntries) {
+      totalNonForcedStyleAndLayoutDuration +=
+        loafEntry.startTime +
+        loafEntry.duration -
+        loafEntry.styleAndLayoutStart;
+
+      for (const script of loafEntry.scripts) {
+        const scriptEndTime = script.startTime + script.duration;
+        if (scriptEndTime < interactionTime) {
+          return;
+        }
+        const intersectingScriptDuration =
+          scriptEndTime - Math.max(interactionTime, script.startTime);
+        totalIntersectingScriptsDuration += intersectingScriptDuration;
+        numScripts++;
+        totalForcedStyleAndLayout += script.forcedStyleAndLayoutDuration;
+        const invokerType = script.invokerType;
+        let subpart: INPSubpart = 'processingDuration';
+        if (script.startTime < interactionTime + inputDelay) {
+          subpart = 'inputDelay';
+        } else if (
+          script.startTime >
+          interactionTime + inputDelay + processingDuration
+        ) {
+          subpart = 'presentationDelay';
+        }
+
+        // Define the record if necessary. Annoyingly TypeScript doesn't yet
+        // recognize this so need a few `!`s on the next two lines to convinced
+        // it is typed.
+        subparts[subpart] ??= {};
+        subparts[subpart]![invokerType] ??= 0;
+        // Increment it with this value
+        subparts[subpart]![invokerType]! += intersectingScriptDuration;
+
+        if (intersectingScriptDuration > slowestScriptDuration) {
+          slowestScriptEntry = script;
+          slowestScriptSubpart = subpart;
+          slowestScriptDuration = intersectingScriptDuration;
+        }
+      }
+    }
+
+    // Gather the loaf summary information into the loafAttribution object
+    const loafSummary: LongAnimationFrameSummary = {
+      numLongAnimationFrames: attribution.longAnimationFrameEntries.length,
+      numIntersectingScripts: numScripts,
+      slowestScript: {
+        entry: slowestScriptEntry,
+        subpart: slowestScriptSubpart,
+        intersectingDuration: slowestScriptDuration,
+        totalDuration: slowestScriptEntry?.duration,
+        compileDuration:
```

Comment:
Using the `?.` optional chaining operator here seems risky because if `slowestScriptEntry` is undefined then you'll end up doing math operations on `undefined`, which will result in `NaN`.

I know there's been some discussion of removing these properties anyway, but if we don't then it's probably better to just omit these (or set them to 0) if there's no slowest script.